components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java

```diff
@@ -73,6 +82,81 @@ public boolean isExecutionForActionTypeEnabled(ActionType actionType) {
+     */
+    public int getHttpClientConnectionPoolSize() {
+
+        int poolSizePropertyValue = DEFAULT_HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY;
+        String poolSizeValue = (String) IdentityConfigParser.getInstance().getConfiguration().
+                get(HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY);
+        if (StringUtils.isNotBlank(poolSizeValue)) {
+            try {
+                poolSizePropertyValue = Integer.parseInt(poolSizeValue);
+            } catch (NumberFormatException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Failed to read Http client connection pool size property in identity.xml." +
+                            " Expects a number. Using the default value: " +
+                            DEFAULT_HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY, e);
+                }
+            }
+        }
+        return poolSizePropertyValue;
+    }
+
+    /**
+     * Retrieves the HTTP read timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP read timeout int value in milliseconds.
+     */
+    public int getHttpReadTimeoutInMillis() {
+
+        return parseTimeoutConfig(HTTP_READ_TIMEOUT_PROPERTY, DEFAULT_HTTP_READ_TIMEOUT_IN_MILLIS);
+    }
+
+    /**
+     * Retrieves the HTTP connection request timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP connection request timeout int value in milliseconds.
+     */
+    public int getHttpConnectionRequestTimeoutInMillis() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_REQUEST_TIMEOUT_PROPERTY,
+                DEFAULT_HTTP_CONNECTION_REQUEST_TIMEOUT_MILLIS);
+    }
+
+    /**
+     * Retrieves the HTTP connection timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP connection timeout int value in milliseconds.
+     */
+    public int getHttpConnectionTimeoutInMillis() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_TIMEOUT_PROPERTY, DEFAULT_HTTP_CONNECTION_TIMEOUT_MILLIS);
+    }
+
+    private int parseTimeoutConfig(String timeoutTypeName, int defaultTimeout) {
+
+        int timeoutPropertyValue = defaultTimeout;
+        String timeoutValue = (String) IdentityConfigParser.getInstance().getConfiguration().get(timeoutTypeName);
+        if (StringUtils.isNotBlank(timeoutValue)) {
+            try {
+                timeoutPropertyValue = Integer.parseInt(timeoutValue);
+            } catch (NumberFormatException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Failed to read " + timeoutTypeName + " property in identity.xml." +
+                            " Expects a number. Using the default value: " + defaultTimeout, e);
+                }
+            }
```

Comment:
Refer the comment above

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/5935/commits/d3435deafaa2baea562aae75e85180c9a5838a7e